src/chainlink/oracle/ChainlinkOracleImpl.sol

```diff
@@ -0,0 +1,264 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {AddressUtils} from "splits-utils/AddressUtils.sol";
+import {AggregatorV3Interface} from "chainlink/interfaces/AggregatorV3Interface.sol";
+import {ConvertedQuotePair, QuotePair, QuoteParams} from "splits-utils/LibQuotes.sol";
+import {QuotePair, QuoteParams} from "splits-utils/LibQuotes.sol";
+import {TokenUtils} from "splits-utils/TokenUtils.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {OracleImpl} from "../../OracleImpl.sol";
+import {ChainlinkPairDetails} from "../../libraries/ChainlinkPairDetails.sol";
+
+/// @title Chainlink Oracle Implementation
+/// @author 0xSplits
+/// @notice A clone-implementation of an oracle using Chainlink
+/// @dev This contract uses token = address(0) to refer to ETH.
+contract ChainlinkOracleImpl is OracleImpl {
+    /// -----------------------------------------------------------------------
+    /// libraries
+    /// -----------------------------------------------------------------------
+
+    using AddressUtils for address;
+    using TokenUtils for address;
+    using ChainlinkPairDetails for mapping(address => mapping(address => PairDetail));
+    using FixedPointMathLib for uint256;
+
+    /// -----------------------------------------------------------------------
+    /// errors
+    /// -----------------------------------------------------------------------
+
+    error InvalidPair_FeedNotSet(QuotePair qp);
+    error StalePrice(AggregatorV3Interface feed, uint256 timestamp);
+    error NegativePrice(AggregatorV3Interface feed, int256 price);
+
+    /// -----------------------------------------------------------------------
+    /// structs
+    /// -----------------------------------------------------------------------
+
+    struct InitParams {
+        address owner;
+        bool paused;
+        SetPairDetailParams[] pairDetails;
+    }
+
+    struct SetPairDetailParams {
+        QuotePair quotePair;
+        PairDetail pairDetail;
+    }
+
+    struct PairDetail {
+        // encoded feed[]
+        bytes path;
+        bool inverted;
+    }
+
+    struct Feed {
+        AggregatorV3Interface feed;
+        /// @dev staleAfter > 1 hours
```

Comment:
`should be > 1 hours`

btw is .. that a reasonable cap for us to put on this? in general like to be ~less opinionated at the contract level unless it's absolutely a terrible terrible idea for anyone to do it. easy enough to suggest values in docs & impose standards in our UI

Reply:
don't feel really strongly about this, I personally wouldn't like to set an oracle that reverts most of the time when there is no volatility so this adds as a good check for us when we are creating the oracle which will most likely be the most used oracle. Can remove the requirement and be more careful about it on our end, fine with both.

Reply:
@mihoward21 wdyt? i recall in the last set of contracts we leaned into maximal freedom in the protocol w strong FE opinions/safe-guards. Does that still feel right to you? I think I still lean that direction personally (going to sleep on it) but looking for another opinion